nixos/modules/services/misc/geoipupdate.nix

```diff
@@ -0,0 +1,84 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.geoipupdate;
+  configFile = pkgs.writeText "geoipupdate.conf" ''
+    AccountID ${cfg.accountID}
+    LicenseKey ${cfg.licenseKey}
+    EditionIDs ${concatStringsSep " " cfg.editionIDs}
+    ${concatStringsSep "\n" (mapAttrsToList (k: v: "${k} ${v}") cfg.extraConfig)}
+  '';
+in {
+  options.services.geoipupdate = with types; {
+    enable = mkEnableOption "the automated GeoIP updater";
+
+    accountID = mkOption {
+      description = "Account ID of your Maxmind account (or 0 for free GeoLite DBs)";
+      type = str;
+      default = "0";
+    };
+
+    licenseKey = mkOption {
```

Comment:
Not sure whether to treat this as a password and use something like `licenseKeyFile` which would be a path to the file containing license key so as not to end up in nix store.

Reply:
Yeah, but their config file format doesn't allow for imports sadly

Reply:
You can generate the final config file in `preStart` (see e.g. `nixos/modules/services/misc/gogs.nix`), though it's ugly.

Reply:
Other possibility would be to add an option which accepts path to the complete configuration file, which the user can place outside of /nix/store if desired.